it/google-cloud-platform/src/test/java/org/apache/beam/it/gcp/storage/GcsResourceManagerTest.java

```diff
@@ -158,6 +165,37 @@ public void testUploadArtifactInvalidLocalPath() {
         IOException.class, () -> gcsClient.uploadArtifact(ARTIFACT_NAME, "/" + UUID.randomUUID()));
   }
 
+  @Test
+  public void testGenerateBucketName_validInput() {
+    String testClassName = "MyTestClass";
+    String runId = "1234567890";
+    String expectedBucketName = "mytestclass-1234567890";
+    String actualBucketName = GcsResourceManager.generateBucketName(testClassName, runId);
+    assertEquals(expectedBucketName, actualBucketName);
+  }
+
+  @Test
+  public void testGenerateBucketName_withSpecialCharactersAndLongName() {
+    String testClassName = "My.Test.Class.With.A.Very.Long.Name";
+    String runId = "1234567890";
+    String expectedBucketName =
+        "my-test-class-with-a-very-long-name-1234567890"; // Periods replaced, truncated
+    String actualBucketName = GcsResourceManager.generateBucketName(testClassName, runId);
+    assertEquals(expectedBucketName, actualBucketName);
+  }
+
+  @Test
+  public void testGenerateBucketName_nullInputs() {
+    assertThrows(
+        NullPointerException.class,
+        () -> GcsResourceManager.generateBucketName("MyTestClass", null) // Null runId
```

Comment:
nit: Prefer

```
() -> GcsResourceManager.generateBucketName("MyTestClass", /* runId= */ null)
```

It's a bit cleaner and works a bit better with the formatter.